hydrolib/core/dflowfm/xyz/parser.py

```diff
@@ -22,13 +22,20 @@ def parse(filepath: Path) -> Dict:
         data: Dict = dict(points=[])
 
         with filepath.open() as f:
-            for line in f.readlines():
+            for linenr, line in enumerate(f.readlines()):
 
-                if len(line) < 5:  # 3 values, two whitespaces
+                line = line.strip()
+                if line.startswith("*") or len(line) == 0:
                     continue
 
-                x, y, z, c = re.split(xyzpattern, line, maxsplit=3)
+                try:
+                    x, y, z, *c = re.split(xyzpattern, line, maxsplit=3)
```

Comment:
Is it required to use `*c` (instead of simply `c`) since we specify `maxsplit=3`? So `c` will contain at most one string?

Reply:
Since the comment (c) is optional, the split might not return a fourth value, then incorrectly raising a ValueError.
When using *c, the fourth value will be a list that may be *empty*, which nicely handles optional comments.